plugins/faustwp/includes/rest/callbacks.php

```diff
@@ -523,3 +534,43 @@ function handle_rest_validate_secret_key_callback( \WP_REST_Request $request ) {
 function rest_validate_secret_key_permission_callback( \WP_REST_Request $request ) {
 	return rest_authorize_permission_callback( $request );
 }
+
+/**
+ * Callback for WordPress register_rest_route() 'callback' parameter.
+ *
+ * Handle POST /faustwp/v1/validate_public_wordpress_url response.
+ *
+ * @link https://developer.wordpress.org/reference/functions/register_rest_route/
+ * @link https://developer.wordpress.org/rest-api/extending-the-rest-api/routes-and-endpoints/#endpoint-callback
+ *
+ * @param \WP_REST_Request $request Current \WP_REST_Request object.
+ *
+ * @return mixed A \WP_REST_Response, or \WP_Error.
+ */
+function handle_rest_validate_public_wordpress_url_callback( \WP_REST_Request $request ) {
+	// Get the frontend URI setting from WordPress.
+	$frontend_uri = faustwp_get_setting( 'frontend_uri' );
+
+	// Retrieve the parameters from the request.
+	$parameters = $request->get_params();
+
+	// Check if the public_wordpress_url parameter is present in the request.
+	if ( isset( $parameters['public_wordpress_url'] ) ) {
+		// Retrieve the value of the public_wordpress_url parameter.
+		$public_wordpress_url = $parameters['public_wordpress_url'];
+
+		// Check if the provided WordPress URL does not match the frontend URI.
+		if ( ! domains_match( $public_wordpress_url, $frontend_uri ) ) {
+			// Return 200 OK if the URLs do not match.
+			$response = new \WP_REST_Response( 'OK', 200 );
+		} else {
+			// Return 400 Bad Request if the URLs match.
+			$response = new \WP_REST_Response( 'Bad Request: Provided URL matches the frontend URI setting.', 400 );
+		}
+	} else {
+		// Return 400 Bad Request if the public_wordpress_url parameter is missing.
+		$response = new \WP_REST_Response( 'Bad Request: public_wordpress_url parameter is missing.', 400 );
```

Comment:
If returning error messages here, why not use them in the JS code instead of hardcoding the error strings there too?

Reply:
Removed those messages.